src/consensus/multiprocess_validation.py

```diff
@@ -71,8 +74,14 @@ def batch_pre_validate_blocks(
                 cost_result = None
             else:
                 if not error and generator is not None and validate_transactions:
+                    gen_refs = NilSerializedProgram
+                    if generator_ref_list is not None and len(generator_ref_list) > 0:
+                        gen_refs = SerializedProgram.from_bytes(generator_ref_list)
                     cost_result = calculate_cost_of_program(
-                        SerializedProgram.from_bytes(generator), constants.CLVM_COST_RATIO_CONSTANT
+                        SerializedProgram.from_bytes(generator),
+                        gen_refs,
+                        constants.CLVM_COST_RATIO_CONSTANT,
+                        False,  # strict_mode=False
```

Comment:
my understanding is that this is correct. When validating blocks on the chain, we accept that there may be some opcodes we don't know about, i.e. not strict mode